### ISSUE TITLE
Error in ATACorrect step

### ISSUE THREAD

**Issue by BenayounLaboratory** [loosolab/TOBIAS#99]
Hi,
sorry if this is an easy issue for python-literate people. We are trying to run TOBIAS on some ATAC-seq data from the lab. Installation worked fine (on python 3.9). 

Here is the command we used: 
`TOBIAS ATACorrect --bam POOLED_OF_PeriMac_AC43_ATACseq.CLEAN.bam --genome /Users/benayoun/Softwares/Genomes/mm10.fa --peaks POOLED_PeriMac_AC43_ATACseq_MACS2_peaks.QC_noMT.bed --prefix OF_PeriMac_AC43_ATACorrect --cores 2`

The run starts seemingly normally, but then it errors out like this:
```
# ----- Input parameters -----
# bam:	POOLED_OF_PeriMac_AC43_ATACseq.CLEAN.bam
# genome:	/Users/benayoun/Softwares/Genomes/mm10.fa
# peaks:	POOLED_PeriMac_AC43_ATACseq_MACS2_peaks.QC_noMT.bed
# regions_in:	None
# regions_out:	None
# blacklist:	None
# extend:	100
# split_strands:	False
# norm_off:	False
# track_off:	[]
# k_flank:	12
# read_shift:	[4, -5]
# bg_shift:	100
# window:	100
# score_mat:	DWM
# prefix:	OF_PeriMac_AC43_ATACorrect
# outdir:	/Volumes/BB_Home_HQ/Immune_sex_dimorphism_Aging/Macrophages/Clean_Macrophage_Data/ATACseq/TOBIAS
# cores:	2
# split:	100
# verbosity:	3


# ----- Output files -----
# /Volumes/BB_Home_HQ/Immune_sex_dimorphism_Aging/Macrophages/Clean_Macrophage_Data/ATACseq/TOBIAS/OF_PeriMac_AC43_ATACorrect_uncorrected.bw
# /Volumes/BB_Home_HQ/Immune_sex_dimorphism_Aging/Macrophages/Clean_Macrophage_Data/ATACseq/TOBIAS/OF_PeriMac_AC43_ATACorrect_bias.bw
# /Volumes/BB_Home_HQ/Immune_sex_dimorphism_Aging/Macrophages/Clean_Macrophage_Data/ATACseq/TOBIAS/OF_PeriMac_AC43_ATACorrect_expected.bw
# /Volumes/BB_Home_HQ/Immune_sex_dimorphism_Aging/Macrophages/Clean_Macrophage_Data/ATACseq/TOBIAS/OF_PeriMac_AC43_ATACorrect_corrected.bw
# /Volumes/BB_Home_HQ/Immune_sex_dimorphism_Aging/Macrophages/Clean_Macrophage_Data/ATACseq/TOBIAS/OF_PeriMac_AC43_ATACorrect_atacorrect.pdf


2021-11-01 13:57:36 (40631) [INFO]	----- Processing input data -----
2021-11-01 13:57:36 (40631) [INFO]	Reading info from .bam file
2021-11-01 13:57:36 (40631) [INFO]	Reading info from .fasta file
2021-11-01 13:57:36 (40631) [INFO]	Processing input/output regions
2021-11-01 13:57:40 (40631) [STATS]	genome: 65 regions | 2730855475 bp | 100.00% coverage
2021-11-01 13:57:40 (40631) [STATS]	input_regions: 108622 regions | 2643565426 bp | 96.80% coverage
2021-11-01 13:57:40 (40631) [STATS]	output_regions: 73981 regions | 102085950 bp | 3.74% coverage
2021-11-01 13:57:40 (40631) [STATS]	peak_regions: 73981 regions | 87290049 bp | 3.20% coverage
2021-11-01 13:57:40 (40631) [STATS]	nonpeak_regions: 108622 regions | 2643565426 bp | 96.80% coverage
2021-11-01 13:57:40 (40631) [STATS]	blacklist_regions: 0 regions | 0 bp | 0.00% coverage
Traceback (most recent call last):
  File "/usr/local/bin/TOBIAS", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/tobias/TOBIAS.py", line 154, in main
    args.func(args)
  File "/usr/local/lib/python3.9/site-packages/tobias/tools/atacorrect.py", line 256, in run_atacorrect
    logger.start_logger_queue()
  File "/usr/local/lib/python3.9/site-packages/tobias/utils/logger.py", line 130, in start_logger_queue
    self.listener.start()
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
  File "/usr/local/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/logging/__init__.py", line 1739, in __reduce__
    raise pickle.PicklingError('logger cannot be pickled')
_pickle.PicklingError: logger cannot be pickled
```

Would you have any pointers as to what is going on and how to fix it?

Thank you!

**Comment by msbentsen** [comment]
Hi,

Thank you for your issue. No worries, this is not super easy and I am also a bit confused about the error to be honest. I can run the test dataset ([test-data](https://github.com/loosolab/TOBIAS/wiki/test-data) and [ATACorrect](https://github.com/loosolab/TOBIAS/wiki/ATACorrect)) it without any issues with `python=3.9.5` and `TOBIAS=0.12.12` on a debian system, so I am wondering if you maybe have a different setup or different versions of other packages.

What type of system are you on? Can you maybe share with me the output of `pip freeze`? Then I will try to reproduce the error. Thank you!

**Comment by BenayounLaboratory** [comment]
Thank you for the quick response and willingness to help!

I solved the issue (not that I understand why) and I will post my solution here if anyone ever encounters a similar issue...

I am working on an OSX/Darwin architecture and was using the "homebrew" for Python. Another python package that I had used stopped working as well yesterday, and after doing some digging on the errors I was getting for the other program, I realized that homebrew automatically upgraded Python to 3.9, which doesn't play well with some previous packages. To make that package work again, I manually downgraded Python to 3.7 again.

I was curious this morning and attempted a clean reinstall of TOBIAS using pip on Python 3.7. It now gets past the error point that I was encountering and is running through the algorithm. Unclear why the downgrade removed the error, or if this is an OSX-only issue.

Thank you again - excited to use the software!
